### PR TITLE
Pallet crowdsale test coverage

### PR DESCRIPTION
**Pull request by JasonTulp**
- Adds Rust tests for crowdsale pallet
- Changes voucher creation to mint all at start into vault account, then transfer to ecah account after
- Other minor changes

### CODE REVIEW

File: pallet/crowdsale/src/lib.rs [hunk 1]
```diff
@@ -203,14 +203,14 @@ pub mod pallet {
 		InvalidCrowdsaleStatus,
 		/// Crowdsale is not enabled
 		CrowdsaleNotEnabled,
+		/// The soft cap price must be greater than zero
+		InvalidSoftCap,
```
Comment (zees-dev) [inline pallet/crowdsale/src/lib.rs:207]: nit: ambigious if softcap total or softcap price
```suggestion
		InvalidSoftCapPrice,
```

File: pallet/crowdsale/src/lib.rs [hunk 2]
```diff
@@ -311,7 +308,7 @@ pub mod pallet {
 		#[transactional]
 		pub fn initialize(
 			origin: OriginFor<T>,
-			payment_asset: AssetId,
+			payment_asset_id: AssetId,
```
Comment (zees-dev) [inline pallet/crowdsale/src/lib.rs:311]: either we append the `_id` suffix to `payment_asset` here and update the `SaleInformation` type attributes to also have `_id` suffix there too (and across the pallet); or remove the `_id` suffix here (and in comment)

File: pallet/crowdsale/src/lib.rs [hunk 3]
```diff
@@ -445,9 +445,9 @@ pub mod pallet {
 			let who = ensure_signed(origin)?;
 
 			// update the sale status if the start block is met
-			SaleInfo::<T>::try_mutate(sale_id, |sale_info: &mut Option<SaleInformation<_, _>>| {
+			SaleInfo::<T>::try_mutate(sale_id, |sale_info| -> DispatchResult {
 				let Some(sale_info) = sale_info else {
-					return Err(Error::<T>::CrowdsaleNotFound);
+					return Err(Error::<T>::CrowdsaleNotFound.into());
 				};
```
Comment (zees-dev) [inline pallet/crowdsale/src/lib.rs:451]: nit: - can apply the same everywhere we do this to reduce lines of code
```suggestion
				let sale_info = sale_info.as_mut().ok_or(Error::<T>::CrowdsaleNotFound)?;
```

File: pallet/crowdsale/src/lib.rs (hunk 1 above)
Comment (JasonTulp) [reply pallet/crowdsale/src/lib.rs]: https://github.com/futureversecom/trn-seed/pull/799/commits/3d396a7ff08f15ec8217f78889223bb0ef24e21c

File: pallet/crowdsale/src/lib.rs (hunk 2 above)
Comment (JasonTulp) [reply pallet/crowdsale/src/lib.rs:311]: https://github.com/futureversecom/trn-seed/pull/799/commits/2fe5ab451abdd359fa0b34a646abb26ecfadebc4

File: pallet/crowdsale/src/tests.rs [hunk 4]
```diff
@@ -51,12 +93,39 @@ fn add_decimals(balance: Balance, decimals: u8) -> Balance {
...
+	}
+}
+
 mod calculate_voucher_rewards {
 	use super::*;
-	use crate::mock::MaxTokensPerCollection;
 
 	#[test]
-	fn calculate_voucher_rewards_works() {
+	fn over_committed_works_1() {
```
Comment (zees-dev) [inline pallet/crowdsale/src/tests.rs:128]: note: could make these tests table driven; tests could potentially be abstracted to a function - which just requires inputs and assertions. Not required though

Comment (JasonTulp) [reply pallet/crowdsale/src/tests.rs:128]: This is a good idea, Maybe at the end if time allows we can do this

**Pull request merged**